From 979e0c1c4a9bab85df58e9253ec9e47e6e54310f Mon Sep 17 00:00:00 2001
From: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>
Date: Fri, 20 Jul 2018 22:24:46 -0700
Subject: [PATCH 1591/1606] ASoC: SOF: set kcontrol cmd for pga widget

set the kcontrol cmd which will be used to send the correct
ipc command to restore volume control value during resume.

Signed-off-by: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>
---
 sound/soc/sof/topology.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/sound/soc/sof/topology.c b/sound/soc/sof/topology.c
index c17bfa1..b3ff74d 100644
--- a/sound/soc/sof/topology.c
+++ b/sound/soc/sof/topology.c
@@ -1129,6 +1129,9 @@ static int sof_widget_load_pga(struct snd_soc_component *scomp, int index,
 	/* get volume control */
 	scontrol = sm->dobj.private;
 
+	/* set cmd for pga kcontrol */
+	scontrol->cmd = SOF_CTRL_CMD_VOLUME;
+
 	/* get topology tlv data */
 	p = kc->tlv.p;
 
-- 
2.7.4

